src/test/scala/is/hail/expr/ir/OrderingSuite.scala

```diff
@@ -219,4 +219,188 @@ class OrderingSuite {
       }
     }
   }
+
+  @Test def testSetContainsOnRandomSet() {
+    val compareGen = Type.genArb
+      .flatMap(t => Gen.zip(Gen.const(TSet(t)), TSet(t).genNonmissingValue, t.genValue))
+    val p = Prop.forAll(compareGen) { case (tset, a, testElem1) =>
+      val telt = tset.elementType
+      val set: Set[Any] = a.asInstanceOf[Set[Any]]
+      val ir = SetContains(GetTupleElement(In(0, TTuple(tset)), 0),
+          GetTupleElement(In(1, TTuple(telt)), 0))
+      val fb = EmitFunctionBuilder[Region, Long, Boolean, Long, Boolean, Boolean]
+
+      Emit(ir, fb)
+
+      val f = fb.result()()
+
+      Region.scoped { region =>
+        val rvb = new RegionValueBuilder(region)
+
+        rvb.start(TTuple(tset))
+        rvb.startTuple()
+        rvb.addAnnotation(tset, set)
+        rvb.endTuple()
+        val doff = rvb.end()
+
+        rvb.start(TTuple(telt))
+        rvb.startTuple()
+        rvb.addAnnotation(telt, testElem1)
+        rvb.endTuple()
+        val k1off = rvb.end()
+
+        if (set.nonEmpty) {
+          val testElem2 = set.head
+
+          rvb.start(TTuple(telt))
+          rvb.startTuple()
+          rvb.addAnnotation(telt, testElem2)
+          rvb.endTuple()
+          val k2off = rvb.end()
+
+          val expected2 = set.contains(testElem2)
+          val actual2 = f(region, doff, false, k2off, false)
+          assert(expected2 == actual2)
+        }
+
+        val expected1 = set.contains(testElem1)
+        val actual1 = f(region, doff, false, k1off, false)
+
+        expected1 == actual1
+      }
+    }
+    p.check()
+  }
+
+  @Test def testDictGetOnRandomDict() {
```

Comment:
I worry about the complexity of this test as a barrier to verifying it is testing what we want.  Can't this done by comparing (1) testing correctness on a few hand-cooked examples, and (2) comparing the compiler and interpreter output a la InterpertSuite?

Reply:
I think the actual comparison logic on this is super simple and a lot of the complexity comes from taking things in and out of regions. would it be better if I just wrapped that ouside of the test?

Reply:
Yeah, we actually duplicate tedious region manipulation logic in various tests and building some nice, common infrastructure for this would certainly improve the testing situation around IR.

Certainly for this test, separate out and make clear the test logic vs. the supporting scaffolding.